apps/contracts/lib/wallet/proxy.js

```
@@ -9,33 +10,46 @@ const walletProxy = require("../contracts/walletProxy");
 const userOperation = require("../constants/userOperations");
 const encodeFunctionData = require("./encodeFunctionData");
 
-module.exports.decryptSigner = (wallet, password) => {
+module.exports.decryptSigner = async (wallet, password, username) => {
   try {
-    const privateKey = AES.decrypt(wallet.encryptedSigner, password).toString(
+    const passwordKey = await generatePasswordKey(password, username)
+    const privateKey = AES.decrypt(wallet.encryptedSigner, Buffer.from(passwordKey).toString('hex')).toString(
       Utf8
     );
     if (!privateKey) return;
-
     return new ethers.Wallet(privateKey);
   } catch (error) {
-    console.error(error);
+    throw(error);
```

Probably just want to leave this to log and swallow the error. I found the aes decrypt function sometimes returns undefined or throws an error with incorrect password. By just logging it we make sure response from this function remains consistent.